sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -132,11 +137,18 @@ private Mono<HttpResponse> attemptAsync(HttpRequest request, boolean eagerlyRead
         }
 
         return nettyRequest.single()
-            .flatMap(response -> {
+            .flatMap(responseAndHeaders -> {
+                HttpResponse response = responseAndHeaders.getT1();
                 if (addProxyHandler && response.getStatusCode() == 407) {
-                    return proxyRetry
-                        ? Mono.error(new ProxyConnectException("Connection to proxy failed."))
-                        : Mono.error(new ProxyConnectException("First attempt to connect to proxy failed."));
+                    if (proxyRetry) {
+                        // Exhausted retry attempt return an error.
+                        return Mono.error(new HttpProxyHandler.HttpProxyConnectException(
+                            "Failed to connect to proxy. Status: 407", responseAndHeaders.getT2()));
```

Comment:
Curious why we can't use `response.getHeaders()` here?

Reply:
This takes the Netty `HttpHeaders` object, not the azure-core `HttpHeaders`.